packages/tpetra/core/test/CrsMatrix/CrsMatrix_PackUnpack.cpp

```diff
@@ -81,24 +81,32 @@ generate_test_matrix (const Teuchos::RCP<const Teuchos::Comm<int> >& comm)
 
   const int world_rank = comm->getRank();
 
-  const int num_row_per_proc = 4;
+  const LO num_row_per_proc = 4;
   Array<GO> row_gids(num_row_per_proc);
 
-  GO start = num_row_per_proc * world_rank;
-  for (LO i = 0; i < num_row_per_proc; ++i) {
+  GO start = static_cast<GO>(num_row_per_proc * world_rank);
+  for (LO i=0; i<num_row_per_proc; ++i) {
     row_gids[i] = static_cast<GO>(start+i);
   }
 
   // Create random, unique column GIDs.
-  const int num_nz_cols = 100;
+  const LO num_nz_cols = 100;
   std::random_device rand_dev;
   std::mt19937 generator(rand_dev());
   std::uniform_int_distribution<GO>  distr(1, 2000);
   std::set<GO> col_gids_set;
+  typedef typename std::set<GO>::size_type SGO;
```

Comment:
Fun fact: `std::set<T>::size_type` is generally the same for all T.  However, your code is still correct.